docs/proposals/scheduling/521-replica-estimator/README.md

```diff
@@ -0,0 +1,176 @@
+
+	// NodeSelector is a selector which must be true for the pod to fit on a node.
+	// Selector which must match a node's labels for the pod to be scheduled on that node.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// If specified, the pod's tolerations.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// ResourceRequest represents the resources required by each replica.
+	// +optional
+	ResourceRequest corev1.ResourceList `json:"resourceRequest,omitempty"`
+}
+```
+
+After the change, the `Object Reference` will be like this. The only difference is converting `corev1.ResourceList` to `*ReplicaRequirements` required by each replica.
+
+```go
+// ObjectReference contains enough information to locate the referenced object inside current cluster.
+type ObjectReference struct {
+	// APIVersion represents the API version of the referent.
+	APIVersion string `json:"apiVersion"`
+
+	// Kind represents the Kind of the referent.
+	Kind string `json:"kind"`
+
+	// Namespace represents the namespace for the referent.
+	// For non-namespace scoped resources(e.g. 'ClusterRole')，do not need specify Namespace,
+	// and for namespace scoped resources, Namespace is required.
+	// If Namespace is not specified, means the resource is non-namespace scoped.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// Name represents the name of the referent.
+	Name string `json:"name"`
+
+	// ResourceVersion represents the internal version of the referenced object, that can be used by clients to
+	// determine when object has changed.
+	// +optional
+	ResourceVersion string `json:"resourceVersion,omitempty"`
+
+	// ReplicaRequirements represents the requirements required by each replica.
+	// +optional
+	ReplicaRequirements *ReplicaRequirements `json:"replicaRequirements,omitempty"`
+
+	// Replicas represents the replica number of the referencing resource.
+	// +optional
+	Replicas int32 `json:"replicas,omitempty"`
+}
+```
+
+### Karmada Scheduler
+
+First, the existing plugins in Karmada Scheduler such as ClusterAffinity, APIInstalled and TaintToleration will select the suitable clusters.
+
+Based on this prefilter result, when assigning replicas, the Karmada Scheduler could try to calculate cluster max available replicas by starting gRPC requests concurrently to the Cluster Accurate Replica Estimator. At last, the Cluster Accurate Replica Estimator will soon return how many available replicas that the cluster could produce. Then the Karmada Scheduler assgin replicas into different clusters in terms of the estimation result.
+
+Furthermore, replica estimation can be considered as a new scheduler plugin.
+We could implement this by modifying function calClusterAvailableReplicas to a interface. The previous estimation method, based on `ResourceSummary` in `Cluster.Status`, is able to be a default normal estimation approach.
+
+### Karmada Cluster Accurate Replica Estimator
```

Comment:
How many clusters serve for each `Cluster Accurate Replica Estimator`?  I mean if there are a huge number of clusters, does it also mean a lot of `Cluster Accurate Replica Estimator`?

Reply:
Now each `Cluster Accurate Replica Estimator` just serves for one cluster, just like `karmada-agent`. If `Cluster Accurate Replica Estimator` servers for multiple clusters, each replica will consume more resource because they have more pod and node informers. I thought one-for-one serving does not have obvious disadvantages and could make the module interaction easier.

Reply:
Ok. I didn't mean it's a flaw, just needs to be annotated.